src/test/java/net/oauth/jsontoken/JsonTokenParserTest.java

```diff
@@ -227,4 +199,47 @@ public void testPublicKey() throws Exception {
       // expected
     }
   }
+
+  private boolean testVerify_timeFrame(Instant issuedAt, Instant expiration) throws Exception {
+    HmacSHA256Signer signer = new HmacSHA256Signer("google.com", "key2", SYMMETRIC_KEY);
+    JsonToken token = new JsonToken(signer, clock);
+    if (issuedAt != null) {
+      token.setIssuedAt(issuedAt);
+    }
+    if (expiration != null) {
+      token.setExpiration(expiration);
+    }
+    token.setAudience("http://www.google.com");
+
+    JsonTokenParser parser = new JsonTokenParser(clock, locators, new IgnoreAudience());
+    JsonToken checkToken = parser.deserialize(token.serializeAndSign());
+    try {
+      parser.verify(checkToken);
+      return true;
+    } catch (IllegalStateException e) {
+      return false;
+    }
+  }
+
+  private void testDeserialize_expectIllegalArgumentException(String tokenString) throws Exception {
+    JsonTokenParser parser = new JsonTokenParser(clock, locators, new IgnoreAudience());
+    try {
+      parser.deserialize(tokenString);
+      fail("Expected IllegalArgumentException");
+    } catch (IllegalArgumentException e) {
+      // expected
+    } catch (IllegalStateException e) {
+      // expected
+    }
```

Comment:
Is IllegalArgument expected or IllegalState?

Reply:
The original tests had both, but I don't think they ever use IllegalArgument. I have removed it.